api/src/functionalTest/java/uk/gov/hmcts/payment/functional/TelephonyPaymentsTest.java

```diff
@@ -0,0 +1,223 @@
+        String status = "success";
+
+        String startDateTime = LocalDateTime.now(DateTimeZone.UTC).toString(DATE_TIME_FORMAT_T_HH_MM_SS);
+
+        dsl.given().userToken(USER_TOKEN)
+            .s2sToken(SERVICE_TOKEN)
+            .returnUrl("https://www.goooooogle.com")
+            .when().createTelephonyPayment(paymentRecordRequest)
+            .then().created(paymentDto -> {
+            String referenceNumber = paymentDto.getReference();
+            assertEquals("payment status is properly set", "Success", paymentDto.getStatus());
+            String endDateTime = LocalDateTime.now(DateTimeZone.UTC).toString(DATE_TIME_FORMAT_T_HH_MM_SS);
+            //update the status
+            dsl.given().userToken(USER_TOKEN)
+                .s2sToken(SERVICE_TOKEN)
+                .returnUrl("https://www.goooooogle.com")
+                .when().updatePaymentStatus(referenceNumber, status)
+                .then().noContent();
+
+            dsl.given().userToken(USER_TOKEN)
+                .s2sToken(SERVICE_TOKEN)
+                .returnUrl("https://www.goooooogle.com")
+                .when()
+                .enableSearch()
+                .searchPaymentsByServiceBetweenDates(paymentRecordRequest.getService(), startDateTime, endDateTime)
+                .then().got(PaymentsResponse.class, paymentsResponse -> {
+                assertTrue("correct payment has been retrieved",
+                    paymentsResponse.getPayments().stream()
+                        .anyMatch(o -> o.getPaymentReference().equals(referenceNumber)));
+                PaymentDto paymentRetrieved = paymentsResponse.getPayments().stream().filter(o -> o.getPaymentReference().equals(referenceNumber)).findFirst().get();
+                assertEquals("correct payment reference retrieved", paymentRetrieved.getCaseReference(), paymentRecordRequest.getReference());
+                assertEquals("payment status is properly set", "Success", paymentRetrieved.getStatus());
+            });
+        });
+    }
+
+    @Test
+    public void retrieveAFailedTelephonyPaymentViaLookup() {
+        String telRefNumber = new Generex("TEL_PAY_\\d{8}").random();
+        PaymentRecordRequest paymentRecordRequest = getTelephonyPayment(telRefNumber);
+        String status = "failed";
+
+        String startDateTime = LocalDateTime.now(DateTimeZone.UTC).toString(DATE_TIME_FORMAT_T_HH_MM_SS);
+
+        dsl.given().userToken(USER_TOKEN)
+            .s2sToken(SERVICE_TOKEN)
+            .returnUrl("https://www.goooooogle.com")
+            .when().createTelephonyPayment(paymentRecordRequest)
+            .then().created(paymentDto -> {
+            String referenceNumber = paymentDto.getReference();
+            assertEquals("payment status is properly set", "Success", paymentDto.getStatus());
+            String endDateTime = LocalDateTime.now(DateTimeZone.UTC).toString(DATE_TIME_FORMAT_T_HH_MM_SS);
+            //update the status
+            dsl.given().userToken(USER_TOKEN)
```

Comment:
some point of time we need to verify callback-url trigger - may be a test subscription to topic etc.